### ISSUE TITLE
Bug: [prefer-nullish-coalescing] Should not report error

### ISSUE THREAD

**Issue by tlebreton** [typescript-eslint/typescript-eslint#10724]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.7.2&fileType=.tsx&code=GYVwdgxgLglg9mABFApgZygCgG4EMA2IKAXImCALYBGKATogD6LgAmKwMYKLAlKedTqIA3gChEiWiighaSPIRSIAhAF5ViAIwBWRADI9iBURXrmYNhy4tEAfiMETpTQG5RAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUFvcgAQBcBPABxQGNoBLEggWhXioDsCB6E6RAM0WjuYImyABZ1yAewCGSZORYBzdFD7QJ0SODABfENqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
function test(value: number | undefined): number {
  return value !== 15 && value !== undefined ? value : 1;
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
        "@typescript-eslint/prefer-nullish-coalescing": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error to report

### Actual Result

Prefer using nullish coalescing operator (`??`) instead of a ternary expression, as it is simpler to read. 2:10 - 2:57

**Comment by FlorinAsavoaie** [comment]
Definitely caused by this [PR](https://github.com/typescript-eslint/typescript-eslint/pull/10517).

**Comment by OlivierZal** [comment]
I’ll take a look

**Comment by OlivierZal** [comment]
I’ll take a look and see if it can be fixed in https://github.com/typescript-eslint/typescript-eslint/pull/10708

**Comment by kirkwaiblinger** [comment]
@OlivierZal I think we might just need to put back the `else { return; }`. I guess it was no longer reached with the existing test cases due to the code changes but apparently is still necessary

https://github.com/typescript-eslint/typescript-eslint/pull/10517/files#diff-dd40826685069133bdddab944ed6d6e71438ba0847484dd9a8ded0faba704283L370-L372

**Comment by kirkwaiblinger** [comment]
Correction - the test cases still reach that, branch but they are able to pass because the `identifierOrMemberExpression` remains unset after leaving the loop. I think that the test case in the issue report breaks because the `value !== undefined` portion of the ternary .test does still fill `identifierOrMemberExpression`, allowing it to continue on to report, even though the rule should bail as soon as it sees the `value !== 15`

**Comment by OlivierZal** [comment]
https://github.com/typescript-eslint/typescript-eslint/pull/10732 should fix
